index.js

```diff
@@ -0,0 +1,314 @@
+/**
+ * Full fhirconvert function: validates, converts, then identifies variables.
+ * @param {string} str - inputted normal syntax expression
+ * @param {Array} vars - array of usable variables entered by user
+ * @returns converted fhirpath expression
+ */
+export function fhirconvert(str, vars) {
+  if (validate(str, vars)) {
+    return varfind(convert(str), vars);
+  } else {
+    return null;
+  }
+}
+
+/**
+ * Verifies normal syntax by confirming var names, function names,
+ * and number of parenthesis.
+ * @param {string} str - inputted normal syntax expression
+ * @param {Array} vars - array of usable variables
+ * @returns boolean, valid or invalidh
+ */
+export function validate(str, vars) {
+  let funs = [
+    "CEILING",
+    "FLOOR",
+    "ABS",
+    "TRUNCATE",
+    "EXP",
+    "SQRT",
+    "LN",
+    "LOG",
+    "ceiling",
+    "floor",
+    "abs",
+    "truncate",
+    "exp",
+    "sqrt",
+    "ln",
+    "log",
+    "and"
```

Comment:
Should this be here?  I don't see it in the convert function, and I don't see "or".

Reply:
Fixed